### DISCUSSION TITLE
is this good PR candidate?

### DISCUSSION THREAD

**Discussion by molleweide** [nvim-treesitter/nvim-treesitter#3455] [Q&A] [answered]
I am working on a tiny util that takes a lua table and returns a ts query. I thought this would be nice for making conditional queries so that you can use table methods instead of having to concat strings etc.

Like this:

```lua
    local q_ = queries.parse({
      field = {
        _name = {
          identifier = {
            "@c_id",
            { "#eq?", "@c_id", "false" },
          },
        },
        _value = {
          false_ = {
            "@c_false",
            {
              "#eq?",
              "@c_false",
              "false",
            },
          },
        },
        { "#eq?", "@c_field", "xxx" },
      },
    })
```

becomes this ts query:

```lisp
(field
  _name:
    (identifier
    ) @c_id (eq @c_id false )

  _value:
    (false_
    ) @c_false (eq @c_false false )

)(eq @c_field xxx )
```

Would this functionality be of interest to `nvim-treesitter`.

**Comment by clason** [comment, marked as answer]
What would be the advantage of the former over the latter?

**Reply by molleweide** [reply]
thanks for replying clason!

good question. these are some observations I make:

1. you can use table methods to compose rather than concat strings to put together large conditional queries. 
2. take advantage of lua table auto formatting. I am not aware if there exists auto formatting for queries?
3. can be convenient over using `string.format` for conditional/dynamic queries in some cases since you can use variables directly instead of mapping them via `%s` .
5. highlighting without having to inject query into strings.

At the end of the day this is subjective I think. I also am by now means as experienced in computer science as you guys are so I am mainly probing to see if this is useful or makes sense, or if I am just reinventing the wheel.

I am going to do testing and more comparisons tbl vs strings and report additional findings I make.
